### DISCUSSION TITLE
Applying different rules for different paths in a single config?

### DISCUSSION THREAD

**Discussion by mfn** [rectorphp/rector#8354] [Q&A]
Is it possible to use a single configuration but have different rules apply to different paths?

I get the impression it doesn't, as `paths` are configured for a single `RectorConfig` and so are rules/sets.

Specifically I'm talking about phpunits "annotations to attributes" converter, which isn't needed on non-tests code. But our non-tests code is significant in size so, ideally those rectors don't run there.

Is it a good/accepted practice to simply use two different configs, one for non-tests and one for tests?

For context: this is about running rector also in CI, where especially with phpunit we want to make sure the same approach is used in all test classes.

Considering a similar tool in spirit, phpstan, there I already had to use different configs and setups but I guess it doesn't hurt to ask :)

thanks!

**Comment by samsonasik** [comment]
In phpunit rules, the is under test classes  check already cover that, see

https://github.com/rectorphp/rector-phpunit/blob/90e87f1032026d2c9c0195a826dc1126d2c40ecd/rules/AnnotationsToAttributes/Rector/ClassMethod/DataProviderAnnotationToAttributeRector.php#L99

**Reply by mfn** [reply]
Ah interesting!

Unfortunately that's just one rector, the other being `AnnotationWithValueToAttributeRector` does not have such check (and it might make sense, as it's very generic).

Thus, I think my question still stands.

**Reply by samsonasik** [reply]
`AnnotationWithValueToAttributeRector` seems doesn't have that check, could you create a PR for it? Thank you.
